flink-cdc-dist/src/main/assembly/assembly.xml

```diff
@@ -0,0 +1,62 @@
+<!--
+Copyright 2023 Ververica Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<assembly>
+    <id>dist</id>
+    <formats>
+        <format>tar.gz</format>
+    </formats>
+    <includeBaseDirectory>true</includeBaseDirectory>
+    <baseDirectory>flink-cdc-connectors-${version}-bin</baseDirectory>
+
+    <dependencySets>
+        <dependencySet>
+            <unpack>false</unpack>
+            <scope>runtime</scope>
```

Comment:
I found some suspicious JARs in the final tarball:
- Awaitility: This should be a test dependency, and should not be a part of the distribution. We need to change its scope to `test` in pom. 
- Hamcrest: This is defined as test scope in out project, but brought into the project as `compile` by Awaitility. I guess it will disappear once we change Awaitility to test scope.
- flink-cdc-dist: This is a JAR without any class file inside. We can just skip it when assembling. 
- commons-cli: What about packaging it directly into `flink-cdc-cli`?

Things missing:
- Log4j JAR and config file: Users cannot have logging locally without them

Reply:
About log4j I made a second thought, that maybe it's better to use log4j JARs under FLINK_HOME directly. We just need to provide a log4j configuration file for printing Flink CDC CLI and composer logs to a local file. Sorry for that!